include/seqan3/core/type_traits/function.hpp

```diff
@@ -23,3 +23,27 @@
  * \returns true or false.
  */
 #define SEQAN3_IS_CONSTEXPR(...) std::integral_constant<bool, __builtin_constant_p((__VA_ARGS__, 0))>::value
+
+// ----------------------------------------------------------------------------
+// multi_lambda
+// ----------------------------------------------------------------------------
+
+namespace seqan3::detail
+{
+
+/*!\brief A type that can conveniently inherit multiple invocables and acts as a union over them.
+ * \tparam invocable_ts The types to inherit from.
+ * \ingroup type_traits
+ */
+template <typename ...invocable_ts>
+struct multi_lambda : invocable_ts...
```

Comment:
Mhmm, multi lambda is not such a good name. I mean it would work with any invocable not just lambdas.
So what about: `multi_invocable` `invocable_pool` `invocable_union`?